internal/driver/common/canvas.go

```diff
@@ -98,7 +98,11 @@ func (c *Canvas) EnsureMinSize() bool {
 
 	ensureMinSize := func(node *RenderCacheNode, pos fyne.Position) {
 		obj := node.obj
-		cache.SetCanvasForObject(obj, c.impl)
+		cache.SetCanvasForObject(obj, c.impl, func() {
```

Comment:
I am concerned that this is running in the rendering thread. If the problem is just happening on SetContent, wouldn't calling Refresh there make sense?

Reply:
It happens long after SetContent.
The problem is that the canvas pixscale is not realised until the window draws - it seems to be around the 3rd time an image is refreshed in the setup process due to configuring and resize etc.

Reply:
Hum, I am wondering if this is a Mac behavior as the sequence for configuring/resizing is OS dependent and I haven't noticed a blurriness even on my screen where scale != 1. I would feel better about this change if it was a Refresh call applied on the entire canvas actually.

Reply:
Yes, this is currently only visible on macOS - that is currently the only OS that sets a pixel scale.
Just to be clear this is not canvas scale.
On a macOS retina screen with scale 1.5, for example:
* Widget size = 20
* Screen coordinate size = 30 // that's size * scale
* Actual pixel size = 60 // that's the screen coordinates * retina scaling (2)

I'm not entirely sure why refreshing the whole canvas is better than objects when they are added to the canvas, but I am pretty sure that would miss the case where objects move canvas, or are added after the initial setup as well. In this case we tell it to update as soon as we know what the pixel scale is.

Reply:
Hmm, what should we do then?
The current state is not ok as we've lost crispness on retina/uhdpi screens so some action must be taken.